src/dataflow/split_samples/script.py

```diff
@@ -0,0 +1,99 @@
+import mudata as mu
+import pandas as pd
+import re
+from pathlib import Path
+
+### VIASH START
+par = {
+  'input': 'harmony_knn/integrated.pynndescent_knn.output',
+  'modality': 'rna',
+  'obs_feature': 'dataset',
+  'output': 'reference_download/sample_split',
+  'drop_obs_nan': "true",
+  'output_compression': None,
+  'output_files': 'reference_download/sample_files.csv'
+}
+
+### VIASH END
+
+# START TEMPORARY WORKAROUND setup_logger
+# reason: resources aren't available when using Nextflow fusion
+# from setup_logger import setup_logger
+def setup_logger():
+    import logging
+    from sys import stdout
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    console_handler = logging.StreamHandler(stdout)
+    logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+    console_handler.setFormatter(logFormatter)
+    logger.addHandler(console_handler)
+
+    return logger
+# END TEMPORARY WORKAROUND setup_logger
+logger = setup_logger()
+
+
+def main():
+    logger.info(f"Reading {par['input']}")
+    input_file = Path(par["input"].strip())
+
+    mdata = mu.read_h5mu(input_file)
+    adata = mdata.mod[par["modality"]]
+
+    logger.info(f"Reading unique features from {par['obs_feature']}")
+    obs_features = adata.obs[par["obs_feature"]].unique().tolist()
+
+    # format obs_features into file compatible names
+    obs_features_s = [s.strip().replace(" ", "_") for s in obs_features]
+    obs_features_s = [re.sub(r'[-\s]', "_", s) for s in obs_features]
+    obs_features_s = [re.sub(r'[^A-Za-z0-9_]', "", s) for s in obs_features]
+
+    # ensure that names are unique, if not append number as suffix
+    if not len(obs_features_s) == len(set(obs_features_s)):
+        logger.info("Ensuring unique names for par['obs_feature']")
+        counts = {}
+        for i, feature in enumerate(obs_features_s):
+            if feature not in counts:
+                counts[feature] = 0
+            else:
+                counts[feature] += 1
+                obs_features_s[i] += "_" + str(counts[feature])
```

Comment:
this chunk was added because the feature names are formatted in L48-L51, to make sure we don't accidentally have non-unique names because of that

Reply:
I know what the intent was, what I meant was that it might be more prudent to not automatically generate unique names here by appending a number and just raising instead.  I think we can a compromise where an argument is introduced that allows generating unique names, which is off by default (the default would raise). I think this is a more robust solution that guards against user error.

Reply:
yea i meant that we may not want raise when errors could be introduced by component processing rather than user error, but introducing another argument would be a good compromise